datalad/local/credentials.py

```diff
@@ -0,0 +1,390 @@
+    virtually no naming restrictions exist (only null-byte and newline are
+    forbidden). However, when naming credentials it is recommended to use
+    simple names in order to enable convenient one-off credential overrides
+    by specifying DataLad configuration items via their environment variable
+    counterparts (see the documentation of the ``configuration`` command
+    for details. In short, avoid underscores and special characters other than
+    '.' and '-'.
+
+    While there are no constraints on the number and nature of credential
+    properties, a few particular properties are recognized on used for
+    particular purposes:
+
+    - 'secret': always refers to the single secret of a credential
+    - 'type': identifies the type of a credential. With each standard type,
+      a list of mandatory properties is associated (see below)
+    - 'last-used': is an ISO 8601 format time stamp that indicated the
+      last (successful) usage of a credential
+
+    Standard credential types and properties
+
+    The following standard credential types are recognized, and their
+    mandatory field with their standard names will be automatically
+    included in a 'get' report.
+
+    - 'user_password': with properties 'user', and the password as secret
+    - 'token': only comprising the token as secret
+    - 'aws-s3': with properties 'key-id', 'session', 'expiration', and the
+      secret_id as the credential secret
+
+    Legacy support
+
+    DataLad credentials not configured via this command may not be fully
+    discoverable (i.e., including all their properties). Discovery of
+    such legacy credentials can be assisted by specifying a dedicated
+    'type' property    """
+    result_renderer = 'tailored'
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify a dataset whose configuration to inspect
+            rather than the global (user) settings""",
+            constraints=EnsureDataset() | EnsureNone()),
+        action=Parameter(
+            args=("action",),
+            nargs='?',
+            doc="""which action to perform""",
+            constraints=EnsureChoice(*credential_actions)),
+        name=Parameter(
+            # exclude from CLI
+            args=tuple(),
+            doc="""name of a credential to set, get, or remove.""",
+            constraints=EnsureStr() | EnsureNone()),
+        spec=Parameter(
+            args=("spec",),
+            doc="""specification of[CMD:  a credential name and CMD]
+            credential properties. Properties are[CMD:  either CMD] given as
+            name/value pairs[CMD:  or as a property name prefixed
+            by a colon CMD].
+            Properties with [CMD: no CMD][PY: a `None` PY] value
+            indicate a property to be deleted (action 'set'), or a
+            property to be entered interactively, when no value is set
+            yet, and a prompt text is given (action 'get').
```

Comment:
I find this piece of the docs for the CMD confusing, and I think I found a bug while trying out what I thought was suggested by it.

I read this doc as if this is the correct way to remove the user name from a stored credential:
```
datalad credentials set nitrc user=    
```

But I think from the command line it should rather be
```
datalad credentials set nitrc :user 
```

right? I think it the doc string (here for the parameter and also for the docsting paragraph on set) should say something like
```suggestion
            Properties [CMD: prefixed with a colon CMD][PY: with a `None` value PY] 
            indicate a property to be deleted (action 'set'), or a
            property to be entered interactively, when no value is set
            yet, and a prompt text is given (action 'get').
```

The command above with the empty value actually did lead to a removal of the ``user`` property though (this is a "legacy" credential, unsure if it matters)
```
datalad credentials set nitrc user= 
nitrc(secret): [:user]
```

I think I find that in line with the docs, but it isn't the intended behavior, right?